frontend/src/components/Pages/FrontPage/MissionOverview/MissionRestartButton.tsx

```diff
@@ -77,9 +78,17 @@ export function MissionRestartButton({ mission }: MissionProps) {
                     <Menu.Item onClick={() => startReRun(ReRunOptions.ReRun)}>
                         {TranslateText('Re-run full mission')}
                     </Menu.Item>
-                    <Menu.Item onClick={() => startReRun(ReRunOptions.ReRunFailed)}>
-                        {TranslateText('Re-run failed and cancelled tasks in the mission')}
-                    </Menu.Item>
+                    {(mission.status === MissionStatus.Failed ||
+                        mission.status === MissionStatus.Aborted ||
+                        mission.status === MissionStatus.Cancelled ||
+                        mission.status === MissionStatus.PartiallySuccessful) &&
+                        mission.tasks.some(
+                            (t) => t.status !== TaskStatus.PartiallySuccessful && t.status !== TaskStatus.Successful
```

Comment:
Why check for mission status and not just check if some tasks are not successful?